components/support/utils/src/main/java/mozilla/components/support/utils/SafeBundle.kt

```diff
@@ -15,31 +15,32 @@ import mozilla.components.support.base.log.logger.Logger
  *
  * See bug 1090385 for more.
  */
-class SafeBundle(private val bundle: Bundle) {
+class SafeBundle(val unsafe: Bundle) {
 
-    @SuppressWarnings("TooGenericExceptionCaught")
-    fun getString(name: String): String? {
-        return try {
-            bundle.getString(name)
-        } catch (e: OutOfMemoryError) {
-            Logger.warn("Couldn't get bundle items: OOM. Malformed?")
-            null
-        } catch (e: RuntimeException) {
-            Logger.warn("Couldn't get bundle items.", e)
-            null
-        }
-    }
+    fun getInt(name: String, defaultValue: Int = 0): Int =
+        safeAccess(defaultValue) { getInt(name, defaultValue) }!!
+
+    fun getString(name: String): String? =
+        safeAccess { getString(name) }
+
+    fun <T : Parcelable> getParcelable(name: String): T? =
+        safeAccess { getParcelable<T>(name) }
 
     @SuppressWarnings("TooGenericExceptionCaught")
-    fun <T : Parcelable> getParcelable(name: String): T? {
+    private fun <T> safeAccess(default: T? = null, block: Bundle.() -> T): T? {
```

Comment:
⭐️ ⭐️ 

First one is for making a nice reusable function. Second is for making it private because that `T` is too powerful.

Also, tests? 🙃

Reply:
Updated with tests